### ISSUE TITLE
Update gulp-sass, gulp-plumber, and gulp-clean-css

### ISSUE THREAD

**Issue by ChocolateLoverRaj** [papercss/papercss#216]
## gulp-sass, gulp-plumber, gulp-clean-css
All of the packages above depend on gulp-util, which is a deprecated package. 
## Update [gulp-clean-css](https://github.com/scniro/gulp-clean-css/pull/50)
gulp-clean-css should be updated to at least v3.9.1, which is anyways being updated in #214 
## Update [gulp-plumber](https://github.com/floatdrop/gulp-plumber/issues/63#issuecomment-458563723)
gulp-plumber should be updated to at least v1.2.
## Update [gulp-sass](https://github.com/dlmanning/gulp-sass/releases/tag/v4.0.0)
gulp-sass should be update to at least v4.0.2.
## This is me asking if I can make a pull request.
Can I make a pull request with these  changes?

**Comment by rhyneav** [comment]
A PR would be great, thank you!

**Comment by TotomInc** [comment]
Gulp is kinda an old tool, I wonder if we can change the build-system to something like webpack that is more "future-proof" as some packages of Gulp are becoming deprecated.

If this migration makes senses, I can work on it.

@rhyneav

**Comment by ChocolateLoverRaj** [comment]
@TotomInc  I'm a little confused what gulp does. I know papercss uses it to compile scss into css, but I don't understand why so many `gulp-something` dependencies are needed. 

Also, why exactly do people who install papercss as an npm package need to install the dependencies and then build it. It seems like many extra steps for people who just want files in `dist/`.

**Comment by TotomInc** [comment]
@ChocolateLoverRaj I think there are many improvements that can be done with the way we use gulp as our build-tool.

I have actually never (except for this project) built a SCSS framework, so I know better the state of build-tools for JS projects.

Gulp have created its own ecosystem. `gulp` itself is like the _core_ library to handle the creation of tasks (creating a development server, bundling source-files into `/dist`, ...). From this, you can use other gulp-dependencies (created by the community) such as [`gulp-sass`](https://www.npmjs.com/package/gulp-sass) to transform your output (in this case, gulp will recognize SASS/SCSS and transpile it into CSS as the browser don't understand SASS/SCSS).

The whole gulp community is kinda old and is slowing becoming deprecated. If we want to maintain this framework, it can be a good idea to change our build-tools to something more "future-proof" with a wide community. Webpack can be a good choice but I'm also thinking of Rollup (don't really know if we can use it for CSS only).

I think we also have some dependencies that should be listed as "dev-dependencies".

**Comment by ChocolateLoverRaj** [comment]
@TotomInc maybe the building can be done _before_ commiting to github, and everything besides the `dist/` folder can be `.npmignored`. That way, there will be no dependencies, users can quickly get the files, and the whole npm package will be smaller in size (unless the min and not css few files are actually larger than the source scss files). The building tools could go on `devDependencies`, and people who want to modify papercss can just fork it, make changes, and build.

**Comment by TotomInc** [comment]
@ChocolateLoverRaj I agree with you for the build-tools that _should_ be listed in `devDependencies`.

For the build and push of the `/dist`, we should do it when we release a new version of the framework (patch, minor or major).

The `/dist` folder _should not be pushed for every commit made_, especially since we are meant to develop new features and fix bugs on the `develop` branch.

When a new version is coming out from the `develop` branch, maintainers should create a versioning commit before merging the `develop` to `master` containing the build of the source-code in the `/dist` folder.

**Comment by ChocolateLoverRaj** [comment]
@TotomInc So what I meant by _before commiting_ was _before publishing a version to npm_. Not in like a pre commit script. 



> I agree with you for the build-tools that _should_ be listed in `devDependencies`.

What exactly do you mean? Do you agree with me that the building shouldn't happen in `postinstall` script?

**Comment by TotomInc** [comment]
Yes, I agree with you, the build of the framework __should not happen__ in the `postinstall` script.

It's our job to provide to the user the latest, pre-built, version of the PaperCSS framework.

**Comment by ChocolateLoverRaj** [comment]
So then everything else can be in `.npmignore`, with no dependencies, right? Also, would the `paper.min.css` file also be included?

**Comment by TotomInc** [comment]
@ChocolateLoverRaj yes the goal is to `.npmignore` unused files so everything related to source-code and build-tools configuration.

The `paper.min.css` would still be distributed in the `/dist`.

I started to work on the gulp migration [in this branch](https://github.com/papercss/papercss/commits/build/remove-gulp). The build is actually working and I've totally removed the need for gulp. I'll open a PR soon when this is available.

@rhyneav

**Comment by ChocolateLoverRaj** [comment]
@TotomInc that's great. I'm guessing it will be papercss 2.0 because it's a breaking change.

**Comment by rhyneav** [comment]
Totally agree that the build should not happen on each install. When publishing to NPM, I've been including the built CSS (and minified) anyway. Same with Github releases, I've just been pasting the built css to the release. I think keeping the built CSS file out of git would be best for now. I can see it being confusing for new contributors, but I'm open to including it if anyone else feels strongly. 

Moving forward out of Gulp-Land and into Webpack-World makes sense given the state of the ecosystem. Thank you for taking that on @TotomInc!

As for versioning, I don't think anything will be breaking (famous last words here...). The end usage of PaperCSS will be the same, it's just installation and development will be improved with Webpack.

**Comment by TotomInc** [comment]
> Moving forward out of Gulp-Land and into Webpack-World makes sense given the state of the ecosystem. Thank you for taking that on @TotomInc!

Well, I've switched the build-tools for something more native to the NodeJS ecosystem. After some research, I found that it's not really possible to bundle a pure-CSS/SCSS framework with webpack, as it always try to add some of its JavaScript inside it.

I used the pure [`sass`](https://www.npmjs.com/package/sass) NodeJS implementation with their API and used mainly [`postcss`](https://github.com/postcss/postcss) to [autoprefix](https://github.com/postcss/autoprefixer) and [`cssnano`](https://github.com/cssnano/cssnano) to minify the compiled CSS. Using PostCSS for a CSS framework makes a lot of sense as their tagline is _PostCSS is a tool for transforming styles with JS plugins._ 🙂 

I'll create a pull-request in the next days with more details on the migration process.

@rhyneav in the meantime, should we close this merge-request?

> As for versioning, I don't think anything will be breaking (famous last words here...).

I agree with you, there will not be any breaking changes as the changes are __internal__ only (build-tool) and doesn't impact the end-user (e.g.: changing some classes names, removing components, ...).

**Comment by TotomInc** [comment]
With #219, this issue is not relevant anymore.
